modelling/src/main/java/org/axonframework/modelling/StateManager.java

```diff
@@ -29,15 +29,50 @@
  * persist them. Implementations may specify whether they load entities through
  * {@link Repository#load(Object, ProcessingContext)} or {@link Repository#loadOrCreate(Object, ProcessingContext)}.
  * <p>
- * Entities are registered by their type in combination with their id. In case an id is given that matches two
- * repositories, the repository that is registered for the most specific type is used. If no repository is found for the
- * given type and id, a {@link MissingRepositoryException} is thrown.
+ * Entities are {@link #register(Repository) registered} by their type in combination with their id. The combination of
+ * {@link Repository#entityType() entity type} and {@link Repository#idType() id type} of all repositories must be
+ * unique and unambiguous. This means you cannot register a repository if another conflicting repository already exists.
+ * If you do, a {@link ConflictingRepositoryAlreadyRegisteredException} will be thrown. Note that superclasses and
```

Comment:
Nit: Although I know it's not introduced in this PR, I am going to ask it nonetheless.
What about we rename `ConflictingRepositoryAlreadyRegisteredException` to `RepositoryAlreadyRegisteredException`? The latter covers the same load and is shorter, so figured this was a fair suggestion.

Reply:
Renamed, I agree